### ISSUE TITLE
CPU Usage spike after saving

### ISSUE THREAD

**Issue by GTziolas** [spyder-ide/spyder#11280]
### What steps will reproduce the problem?

After typing anything inside a .py file and pressing ctrl+s to save or simply saving it causes a 1 sec delay and a CPU Usage spike of about 35%. (Intel Core i7- 4770)

## Versions

* Spyder version: 4.0.1 
* Python version: 3.7.0
* Qt version: 5.9.6
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```
atomicwrites >=1.2.0         :  1.3.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
cloudpickle >=0.5.0          :  1.2.2 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
IPython >=4.0                :  7.9.0 (OK)
jedi =0.14.1                 :  0.14.1 (OK)
nbconvert >=4.0              :  5.6.0 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
psutil >=0.3                 :  5.6.3 (OK)
pygments >=2.0               :  2.4.2 (OK)
pylint >=0.25                :  2.4.3 (OK)
pyls >=0.31.2;<0.32.0        :  0.31.2 (OK)
zmq >=17                     :  18.1.0 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
rtree >=0.8.3                :  0.8.3 (OK)
sphinx >=0.6.6               :  2.2.1 (OK)
spyder_kernels >=1.8.1;<2.0.0:  1.8.1 (OK)
watchdog                     :  None (OK)
cython >=0.21                :  0.29.13 (OK)
matplotlib >=2.0.0           :  3.1.1 (OK)
numpy >=1.7                  :  1.17.2 (OK)
pandas >=0.13.1              :  0.25.2 (OK)
scipy >=0.17.0               :  1.3.1 (OK)
sympy >=0.7.3                :  1.4 (OK)
```

**Comment by ccordoba12** [comment]
Please share your file because your description is not enough to track the problem.

**Comment by DaveDarrel** [comment]
Big memory problems each time I play around with the new Spyder 4.0.1. Ex: After launching Deep Neural Nets (tensorflow 2.0) a few times, the computer start to freeze until I reboot Spyder :( Never had this memory problems with V3 before (Windows 10, with Anaconda 1.9.7).

Executing this code:

model = Sequential()
model.add(Dense(X_train.shape[1], 
                kernel_initializer = 'uniform', 
                input_dim = X_train.shape[1])) #INPUT LAYER
#BatchNormalization(),
ReLU(),
#Dropout(0.1),
model.add(Dense(X_train.shape[1]*2)) #HIDDEN LAYER
ReLU(),
model.add(Dense(X_train.shape[1])) #HIDDEN LAYER
BatchNormalization(),
ReLU(),
Dropout(0.1),
model.add(Dense(16, kernel_initializer = 'uniform', activation='linear')) #OUTPUT LAYER
    
# Compile the model with MAE loss
model.compile(tf.keras.optimizers.Adam(lr=0.0035, amsgrad=True),
              loss='mean_absolute_error')

# Fit the model
history = model.fit(X_train, y_train,
                    batch_size=10,
                    epochs=100,
                    #validation_split=0.15,
                    verbose=1)


pred_scaled = pd.DataFrame(y_scaler.inverse_transform(model.predict(X_test)))
plt.plot(pred_scaled.T)
plt.plot(data_all)
plt.legend()

**Comment by GTziolas** [comment]
Uninstalling the KITE plugin seems to have fixed the problem

**Comment by ccordoba12** [comment]
> Uninstalling the KITE plugin seems to have fixed the problem

Ok, thanks for letting us know about it. Since we're in contact with Kite developers, could you share with us the code that's causing you this problem? That way we could inform them to take a look at it and try to fix it.

**Comment by DaveDarrel** [comment]
Update: 

KITE wasn't installed. 

After a couple of tries (launched multiple scripts that I lauched with 3.3.6 before), 4.0.1 start to be very very slow after a couple of executions. Forced to restart computer a couple of times. Can't work with it... same problem each time I try. Forced to go back to 3.3.6 ! Hope the CPU usage spike problem will be found soon ! 

Thanks !

**Comment by ccordoba12** [comment]
> 4.0.1 start to be very very slow after a couple of executions

So is it code execution what is slow in your case? Please notice that is issue is about saving a file in the editor, which causes high CPU usage, not code execution.

**Comment by ccordoba12** [comment]
Closing due to lack of response.
